pkg/providers/ovirt/mappings/merger.go

```diff
@@ -0,0 +1,83 @@
+package mappings
+
+import (
+	v2vv1alpha1 "github.com/kubevirt/vm-import-operator/pkg/apis/v2v/v1alpha1"
+	"github.com/kubevirt/vm-import-operator/pkg/utils"
+)
+
+// MergeMappings creates new resource mapping spec containing all of the mappings from the externalMappingSpec with all mappings from crMappings. Mappings from the crMappings will overwrite ones from the external mapping.
+func MergeMappings(externalMappingSpec *v2vv1alpha1.ResourceMappingSpec, vmiMapping *v2vv1alpha1.OvirtMappings) v2vv1alpha1.OvirtMappings {
+	if externalMappingSpec == nil && vmiMapping == nil {
+		return v2vv1alpha1.OvirtMappings{}
+	}
+	primaryMappings, secondaryMappings := extractMappings(externalMappingSpec, vmiMapping)
+
+	networkMappings := mergeMappings(primaryMappings.NetworkMappings, secondaryMappings.NetworkMappings)
+	storageMappings := mergeMappings(primaryMappings.StorageMappings, secondaryMappings.StorageMappings)
+
+	ovirtMappings := v2vv1alpha1.OvirtMappings{
+		NetworkMappings: networkMappings,
+		StorageMappings: storageMappings,
+	}
+	return ovirtMappings
+}
+
+func extractMappings(externalMappingSpec *v2vv1alpha1.ResourceMappingSpec, crMappings *v2vv1alpha1.OvirtMappings) (primaryMappings *v2vv1alpha1.OvirtMappings, secondaryMappings *v2vv1alpha1.OvirtMappings) {
+	if crMappings == nil {
+		primaryMappings = &v2vv1alpha1.OvirtMappings{}
+	} else {
+		primaryMappings = crMappings
+	}
+
+	if externalMappingSpec == nil {
```

Comment:
since `ResourceMappingSpec.OvirtMappings` is optional , I'd suggest to check that it isn't nil so condition will look like:
`if externalMappingSpec == nil || externalMappingSpec.OvirtMappings == nil`

Reply:
Good catch! Thanks!